pkg/manifest/os.go

```diff
@@ -648,14 +649,10 @@ func (p *OS) serialize() osbuild.Pipeline {
 		}
 		pipeline.AddStage(osbuild.NewSystemdUnitCreateStage(regServiceStageOptions))
 		p.EnabledServices = append(p.EnabledServices, subscribeServiceFile)
+	}
 
-		if rhsmConfig, exists := p.RHSMConfig[subscription.RHSMConfigWithSubscription]; exists {
-			pipeline.AddStage(osbuild.NewRHSMStage(rhsmConfig))
-		}
-	} else {
-		if rhsmConfig, exists := p.RHSMConfig[subscription.RHSMConfigNoSubscription]; exists {
-			pipeline.AddStage(osbuild.NewRHSMStage(rhsmConfig))
-		}
+	if rhsmConfig := p.RHSMConfig; rhsmConfig != nil {
```

Comment:
Maybe just:
```go
diff --git a/pkg/manifest/os.go b/pkg/manifest/os.go
index c0d238533..2cb4d3e5d 100644
--- a/pkg/manifest/os.go
+++ b/pkg/manifest/os.go
@@ -651,8 +651,8 @@ func (p *OS) serialize() osbuild.Pipeline {
                p.EnabledServices = append(p.EnabledServices, subscribeServiceFile)
        }
 
-       if rhsmConfig := p.RHSMConfig; rhsmConfig != nil {
-               pipeline.AddStage(osbuild.NewRHSMStage(osbuild.NewRHSMStageOptions(rhsmConfig)))
+       if p.RHSMConfig != nil {
+               pipeline.AddStage(osbuild.NewRHSMStage(osbuild.NewRHSMStageOptions(p.RHSMConfig)))
        }
 
        if waConfig := p.WAAgentConfig; waConfig != nil {

```

Reply:
Yes, I probably just edited one of the original conditions without thinking about it too much :facepalm: Thanks!

Reply:
I simplified the condition (and a few additional ones that looked the same) in a separate commit.